releng/jenkins/release-prepare-branches/Jenkinsfile

```diff
@@ -0,0 +1,236 @@
+            error('XTEXT_VERSION invalid')
+          }
+          currentBuild.displayName = "#${env.BUILD_NUMBER} ${xtextVersion}";
+          if (params.DRY_RUN) {
+            currentBuild.displayName += " (Dry Run)"
+          }
+
+          println "xtext version to be released ${xtextVersion}"
+          println "branch to be created ${branchName}"
+          println "tag to be created ${tagName}"
+
+          sshagent([CREDENTIAL_ID_GENIE_XTEXT_GITHUB]) {
+            wrap([$class: 'BuildUser']) {
+              repositoryNames.split(',').each {
+                sh """
+                  git clone -b ${params.SOURCE_BRANCH} --depth 1 ${baseGitURL}/${it}.git
+                  cd ${it}
+                  git config user.name "${env.BUILD_USER}"
+                  git config user.email "${env.BUILD_USER_EMAIL}"
+                  git checkout -b ${branchName}
+                  cd ..
+                """
+              }
+            }
+          } // END sshagent
+        } // END script
+      } // END steps
+    } // END stage
+    
+    stage('Modify') {
+      steps {
+        script {
+          repositoryNames.split(',').each {
+            print "##### Preparing $it ########"
+            dir(it) {
+              modifyFiles (it, xtextVersion, branchName)
+            }
+          }
+        } // END script
+      } // END steps
+    } // END stage
+
+
+    stage('Commit & Push') {
+      steps {
+        script {
+          repositoryNames.split(',').each {
+            dir (it) {
+              // Avoid commit when no change has happened. This would make the pipeline fail.
+              sh label: "Commit changes", script: """
+                echo "#### Commit changes for ${it} ####"
+                git diff-index --quiet HEAD || git commit --signoff -a --allow-empty -m '[release] version $xtextVersion'
+                git tag --force -a ${tagName} -m 'release ${tagName}'
+              """
+              if (params.DRY_RUN || params.VERBOSE) {
+                sh "git diff ${params.SOURCE_BRANCH}..${branchName}"
+              }
+            }
+          }
+          if (!params.DRY_RUN) {
+            withCredentials([string(credentialsId: "${GITHUB_API_CREDENTIALS_ID}", variable: 'GITHUB_API_TOKEN')]) {
+            sshagent([GENIE_XTEXT_CREDENTIALS]) {
+              repositoryNames.split(',').each {
+                dir (it) {
+                  sh """
+                    git push --force --tags origin ${branchName}
+                  """
+                }
+              }
+            }}
+            slackSend message: "RELEASE BRANCH '${branchName}' PREPARED.", botUser: true, channel: 'xtext-builds', color: '#00FF00'
+          }
+        } // END script
+      } // END steps
+    } // END stage
+  } // END stages
+
+} // END pipeline
+
+
+/**
+ * Retrieve the Xtext version from 'versions.gradle' of the branch from xtext-lib
+ */
+def getCurrentXtextVersion (branch) {
+  // TODO replace 'master' in curl url
```

Comment:
replace master with what?

Reply:
I had a problem with the "branch" variable getting inserted. Since there were more urgent things to do, I left that as a comment. And then forgot to tackle this again. Solved now.